pkg/build/api/types.go

```diff
@@ -20,6 +20,30 @@ const (
 	DefaultDockerLabelNamespace = "io.openshift."
 )
 
+const (
+	// SecretTypeSource contains data needed to authenticate against SCM servers.
+	//
+	// SecretTypeSource keys are devided into these option groups:
+	//   Basic authentication group. Requires at least one of fields:
+	//     - Secret.Data["username"] - username used to authenticate against SCM server
+	//     - Secret.Data["password"] - password or token needed to authenticate against SCM server
+	//     - Secret.Data["ca.crt"] - CA certificate needed to authenticate against SCM server
+	//   SSH authentication group. Required field:
+	//     - Secret.Data["ssh-privatekey"] - private SSH key needed to authenticate against SCM server
+	SecretTypeSource kapi.SecretType = "Source"
```

Comment:
> And if we define those types they should be defined upstream under the Kube namespace

The thought was that we would allow expanding the upstream types via a validator plugin.